auth/client/src/main/java/org/wildfly/security/auth/client/WildFlyClientResolverProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.client;
+
+import org.kohsuke.MetaInfServices;
+import org.wildfly.client.config.ResolverProvider;
+
+/**
+ * Implementation of the ResolverProvider interfact that allows another project
+ * to use Functions from Encrypted Expression Resolver without adding an
+ * Elytron dependency.
+ * @author <a href="mailto:prpaul@redhat.com">Prarthona Paul</a>
+ */
+
+@MetaInfServices(value = ResolverProvider.class)
+public class WildFlyClientResolverProvider implements ResolverProvider{
```

Comment:
Can you provide some more details on what this class is used for?

Reply:
Sure! This is part of the service loader process. 
The ResolverProvider interface lives inside the wildfly-client-config project and the WildFlyClientResolverProvider class implements that class in wildfly-elytron to give access to the classes and functions needed to resolve and decrypt the encrypted expression.